### ISSUE TITLE
[BUG] [Formatter] Refactor inline conditions

### ISSUE THREAD

**Issue by sondrelg** [Riverside-Healthcare/djLint#182]
- [x] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [x] I've searched the [issues](https://github.com/Riverside-Healthcare/djLint/issues)
- [x] I've read the [docs](https://djlint.com)

## System Info
 - Python: 3.10
 - djlint: 0.7.4
 - template language: jinja

## Issue

All conditions seem to be reformatted - even inline conditions that matter for string formatting:

```diff
❯ djlint test.html --reformat --indent 2

Reformatting 1/1 files ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 00:00    


test.html
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
@@ -1 +1,2 @@

-Dog{% if another_condition != 1 %}s{% endif %}
+Dog
+{% if another_condition != 1 %}s{% endif %}

1 file was updated.
```

In this example, the HTML goes from displaying `Dogs` to `Dog s`. Would it be possible to only refactor conditions on column index zero?

## How To Reproduce

- `echo "Dog{% if another_condition != 1 %}s{% endif %}" > test.html`
- `djlint --reformat test.html`

**Comment by christopherpickering** [comment]
Thanks, the trick is to know when the {% if %} is intentionally "inline", like div vs span (for example from prettier):
![image](https://user-images.githubusercontent.com/17788706/153881603-de33d187-3e07-4178-b1c5-4b8c267ea1b0.png)

I don't think we can use the line index, but maybe there is something else we can use. Maybe a regex ``[a-z]{%`` or something like this, for cases to ignore.

**Comment by sondrelg** [comment]
I think for our case, it's fair to say that a condition is intentionally inlined if the `{%` tag _doesn't_ start on column index 0, where the column index 0 means it's the first non-whitespace character on a line, or it's the first character after an opening HTML tag. 

In other words, this is intentionally inline
```html
Dog{% if condition %}s{% endif %}
```

and this is not
```html
{% if condition %}s{% endif %}
```

Similarly, this is intentional
```html
<p>Dog{% if condition %}s{% endif %}</p>
```

and this is not 
```html
<p>{% if condition %}s{% endif %}</p>
```

Not sure if there are other cases to consider that contradict this logic, but that's my first though 👍

**Comment by christopherpickering** [comment]
Yeah, its a hard problem to solve. What if, for example, someone wants to format minified html? In all our example so far, if we minify it, everything will be on a single line.  So nothing is on index 0. If you are starting with "pre-formatted" html then you just need an indenter like djhtml, right?

I'm thinking in general that `{%` almost never comes after a alpha character unless it is supposed to stay there. The more I see this the more I like the jinja  and njk `{%-` tag 😀 I wish dj would pick that up!

**Comment by sondrelg** [comment]
Yeah the regex would certainly fix it in that sense. Maybe matching on any alphanumeric previous character, not counting whitespace would be more robust, to also handle examples like `0 dog{% if cond %}s`?
